front/src/applications/stdcm/hooks/useLinkedPathSearch.ts

```diff
@@ -0,0 +1,170 @@
+    useOsrdConfActions() as StdcmConfSliceActions;
+
+  const timetableID = useSelector(getTimetableID);
+  const searchDatetimeWindow = useSelector(getSearchDatetimeWindow);
+
+  const selectableSlot = useMemo(() => {
+    const startDate = searchDatetimeWindow ? new Date(searchDatetimeWindow.begin) : new Date();
+    startDate.setUTCHours(0, 0, 0);
+    return {
+      start: startDate,
+      end: searchDatetimeWindow?.end || startDate,
+    };
+  }, [searchDatetimeWindow]);
+
+  const [linkedPathDate, setLinkedPathDate] = useState<Date>(selectableSlot.start);
+  const [displaySearchButton, setDisplaySearchButton] = useState(true);
+  const [linkedPathID, setLinkedPathID] = useState<string>();
+  const [linkedPathResults, setLinkedPathResults] = useState<StdcmLinkedPathResult[]>([]);
+
+  const searchPayload = useMemo(
+    () =>
+      ({
+        object: 'trainschedule',
+        query: [
+          'and',
+          ['search', ['train_name'], linkedPathID],
+          ['=', ['timetable_id'], timetableID!],
+        ],
+      }) as SearchPayload,
+    [timetableID, linkedPathID]
+  );
+
+  const operationalPointSearch = useCallback(
+    async ({ originID, destinationID }: { originID: string; destinationID: string }) => {
+      try {
+        const payloadOP = {
+          object: 'operationalpoint',
+          query: ['or', ['=', ['obj_id'], originID], ['=', ['obj_id'], destinationID]],
+        };
+        const resultsOP = await postSearch({ searchPayload: payloadOP, pageSize: 25 }).unwrap();
+        const groupedResults = groupBy(resultsOP, 'obj_id');
+        return {
+          origin: groupedResults[originID][0],
+          destination: groupedResults[destinationID][0],
+        };
+      } catch (error) {
+        console.error('Failed to fetch operational points:', error);
+        return {};
+      }
+    },
+    [postSearch]
+  );
+
+  const launchTrainScheduleSearch = useCallback(async () => {
+    setDisplaySearchButton(false);
+    try {
+      const results = (await postSearch({
+        searchPayload,
+        pageSize: 25,
+      }).unwrap()) as SearchResultItemTrainSchedule[];
+      const filteredResults = results.filter((result) =>
+        isEqualDate(linkedPathDate, new Date(result.start_time))
+      );
+
+      if (!filteredResults.length) {
+        setDisplaySearchButton(true);
+        return;
+      }
+
+      const newLinkedPathResults = await Promise.all(
+        filteredResults.map(async (result) => {
+          const opList = result.path as PathItemOperationalPoint[];
+          const opIds = {
+            originID: opList.at(0)!.operational_point,
+            destinationID: opList.at(-1)!.operational_point,
+          };
+          const opDetails = (await operationalPointSearch(opIds)) as {
+            origin: SearchResultItemOperationalPoint;
+            destination: SearchResultItemOperationalPoint;
+          };
```

Comment:
You can move some logic in getExtremitiesDetails 
```suggestion
          const opDetails = await getExtremitiesDetails(result.path);
```